maven.grammar/src/org/netbeans/modules/maven/hyperlinks/HyperlinkProviderImpl.java

```diff
@@ -453,5 +215,331 @@ private FileObject getPath(FileObject parent, String path) {
+                    if (token != null && token.id().equals(XMLTokenId.TAG) && TokenUtilities.equals(token.text(), ">")) {//NOI18N
+                        xml.movePrevious();
+                        token = xml.token();
+                        if (token != null && token.id().equals(XMLTokenId.TAG)) {
+                            if (TokenUtilities.equals(token.text(), "<module")) {//NOI18N
+                                if (!ftext.endsWith("/pom.xml")) {
+                                    ftext += "/pom.xml"; //NOI18N
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    xml.movePrevious();
+                    token = xml.token();
+                    if (token != null && token.id().equals(XMLTokenId.TAG) && TokenUtilities.equals(token.text(), ">")) { //NOI18N
+                        xml.movePrevious();
+                        token = xml.token();
+                        if (TokenUtilities.equals(token.text(), "<artifactId") || //NOI18N
+                            TokenUtilities.equals(token.text(), "<groupId") || //NOI18N
+                            TokenUtilities.equals(token.text(), "<type") || //NOI18N
+                            (TokenUtilities.equals(token.text(), "<version") && !ftext.startsWith("${"))) { //NOI18N
+                            resetSequenceToDependencyTagToken(xml);
+                            if (TokenUtilities.equals(xml.token().text(), "<dependency")) {
+                                while(!TokenUtilities.equals(xml.token().text(), "</dependency")) { //NOI18N
+                                    xml.moveNext();
+                                    token = xml.token();
+                                    if (TokenUtilities.equals(token.text(), "<artifactId")) { //NOI18N
+                                        xml.moveNext();
+                                        xml.moveNext();
+                                        token = xml.token();
+                                        artifactId = token.text().toString();
+                                        xml.moveNext();
+                                    } else if (TokenUtilities.equals(token.text(), "<groupId")) { //NOI18N
+                                        xml.moveNext(); 
+                                        xml.moveNext();
+                                        token = xml.token();
+                                        groupId = token.text().toString();
+                                        xml.moveNext();
+                                    } else if (TokenUtilities.equals(token.text(), "<version")) { //NOI18N
+                                        xml.moveNext(); 
+                                        xml.moveNext();
+                                        token = xml.token();
+                                        if (token.text().toString().startsWith("${")) { //NOI18N
```

Comment:
here, using TokenUtilities.startsWith() could help - if the token is long. Or save the String instance so it does not materialize on line 360 again.